test.sh

```diff
@@ -1,21 +1,27 @@
 #!/usr/bin/env bash
```

Comment:
You always want to run `bash` with error checking enabled otherwise you will have many errors hidden and/or obscured otherwise (and you do have some in this code). Take a look at some of the scripts in other repos on how to construct this line so that you get this functionality.

Reply:
added `set -eo pipefall` at the top of the script to make sure the script immediately exists on a failure

Reply:
`!/usr/bin/env bash -ex` instead

Reply:
The `set -ex` works but don't forget to add `set -o pipefail`